data/campaigns/Descent_Into_Darkness/utils/abilities.cfg

```diff
@@ -128,109 +128,21 @@
-        {DID_SHADOW_VEIL_HANDLER}
-    [/dummy]
-#enddef
-
-#define ABILITY_SHADOW_VEIL_HELPER SIDE
-    [hides]
-        id=did_shadow_veil_helper
-        [filter]
-            [filter_location]
-                radius=5
-                [filter]
-                    ability=did_shadow_veil
-                    [filter_side]
-                        [allied_with]
-                            side={SIDE}
-                        [/allied_with]
-                    [/filter_side]
-                [/filter]
-            [/filter_location]
-        [/filter]
-    [/hides]
-#enddef
-
-#define DID_SHADOW_VEIL_HANDLER
-    [event]
-        id=did_shadow_veil_unit_placed
-        name=unit placed
-        first_time_only=no
-        [filter]
-            side=1
-            race=undead
-            [not]
-                ability=did_shadow_veil_helper
-            [/not]
-        [/filter]
-        [filter_condition]
-            [have_unit]
-                ability=did_shadow_veil
-                side=1
-            [/have_unit]
-        [/filter_condition]
-        [modify_unit]
-            [filter]
-                x,y=$x1,$y1
-                [not]
-                    ability=did_shadow_veil_helper
-                    [or]
-                        ability=did_shadow_veil
-                    [/or]
-                [/not]
-            [/filter]
-            [object]
-                duration=scenario
-                [effect]
-                    apply_to=new_ability
-                    [abilities]
-                        {ABILITY_SHADOW_VEIL_HELPER $unit.side}
-                    [/abilities]
-                [/effect]
-            [/object]
-        [/modify_unit]
-    [/event]
-    [event]
-        id=did_shadow_veil_malkeshar
-        name=post advance
-        first_time_only=yes
-        [filter]
-            type=Mal Keshar
-            ability=did_shadow_veil
-        [/filter]
-        [modify_unit]
+        affect_self=no
+        affect_allies=yes
+        [affect_distant]
+            radius=5
             [filter]
                 race=undead
-                [not]
-                    ability=did_shadow_veil_helper
-                    [or]
-                        ability=did_shadow_veil
-                    [/or]
-                [/not]
-                [filter_side]
-                    [allied_with]
-                        side=$unit.side
-                    [/allied_with]
-                [/filter_side]
             [/filter]
-            [object]
-                take_only_once=no
-                duration=scenario
-                [effect]
-                    apply_to=new_ability
-                    [abilities]
-                        {ABILITY_SHADOW_VEIL_HELPER $this_unit.side}
-                    [/abilities]
-                [/effect]
-            [/object]
-        [/modify_unit]
-    [/event]
+        [/affect_distant]
+    [/hides]
```

Comment:
I assume you have tested this in DiD and it still works?

Reply:
Yes, it's work perfectly